sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -255,9 +255,10 @@ class HostKernel : public HostKernelBase {
   runOnHost(const NDRDescT &NDRDesc) {
     sycl::range<Dims> GroupSize;
     for (int I = 0; I < Dims; ++I) {
+      if (NDRDesc.LocalSize[I] == 0 ||
+          NDRDesc.GlobalSize[I] % NDRDesc.LocalSize[I] != 0)
+        throw runtime_error("Invalid local size for global size");
```

Comment:
It took me some time to understand the confusion... I have the feeling you are throwing a `std::runtime_error` and not the `sycl::runtime_error`.

Reply:
Unqualified name lookup should pick up `cl::sycl::runtime_error` unless someone throws a `using namespace std` inside `cl::sycl::detail`, which is against style rules anyways (especially inside a header).

Reply:
Sure, but it is easy to make the code clearer here.